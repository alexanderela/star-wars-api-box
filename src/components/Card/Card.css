@import url("https://fonts.googleapis.com/css?family=Pathway+Gothic+One");

.Card {
  background: url('../../assets/swapi_rough_gray.jpg');
  border-radius: 50px;
  color: white;
  cursor: pointer;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 1.1rem;
  height: 20rem;
  letter-spacing: 0.1rem;
}

.Card:hover {
  box-shadow: #d9ffd9 0px 0px 15px 2px;
}

.Card-active {
	box-shadow: #d9ffd9 0px 0px 15px 2px;
}

.fav-btn-card-container {
	align-items: center;
	display: flex;
	justify-content: space-between;
	padding: 0.5rem 2rem;
}

.fav-btn {
	align-items: center;
	border-radius: 50%;
	box-shadow: ;
	cursor: pointer;
	display: flex;
	height: 3.5rem;
	justify-content: center;
	width: 3.5rem;
}

.fav-btn:hover {
 	border-radius: 50%;
  box-shadow: #d9ffd9 0px 0px 20px 2px;
}

.fav-btn:focus {
  outline: 0;
}


.fav-btn-inactive {
	background-color: #FFF;
	border-radius: 50%;
}

.fav-btn-active {
	background-color: #ffdf00;
	border-radius: 50%;
}

.fa-jedi {
	color: #000;
	font-size: 2rem;
}

.card-text {
	margin: 0.5rem 0.75rem;
}

.card-header {
  color: #f9d500;
 	font-weight: 600;
}
