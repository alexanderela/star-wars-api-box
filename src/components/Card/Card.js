import React from 'react';
import './Card.css';
import PropTypes from 'prop-types';

const Card = ({
  entry, toggleFavorites, favorites, isFavorite,
}) => {
  const cardProperties = entry.properties.map(property => (
    <p key={property.header} className="card-text">
      <span className="card-header">
        {property.header}
      </span>
      {property.text}
    </p>
  ));

  return (
    <div className="Card">
      <div className="fav-btn-card-container">
        <h3>{entry.name}</h3>
        <button
          type="submit"
          className={`fav-btn 
						${(isFavorite || favorites.includes(entry))
            ? 'fav-btn-active'
            : 'fav-btn-inactive'}`}
          onClick={() => toggleFavorites(entry)}
        >
          <i className="fas fa-jedi" />
        </button>
      </div>
      {cardProperties}
    </div>
  );
};

Card.propTypes = {
  entry: PropTypes.object.isRequired,
  toggleFavorites: PropTypes.func.isRequired,
  favorites: PropTypes.array.isRequired,
  isFavorite: PropTypes.bool.isRequired,
};

export default Card;
