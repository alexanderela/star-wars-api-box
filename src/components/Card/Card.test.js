/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import { shallow, mount } from 'enzyme';
import people from '../../mockData/mockPeople.js';

describe('Card component', () => {
	let wrapper;
	let mockFavs;
	let mockEntry;

	const selectCardFunc = jest.fn()
	const mockToggleFavs = jest.fn()
	const mockPerson = {
	          "name": "Luke Skywalker",
	          "id": "Luke Skywalker",
	          "isFavorite": false, 
	          "type":"people",
	          "properties": [
							{"header": 'Homeworld: ', "text": "Tatooine"},
							{"header": 'Species: ', "text": "human"},
							{"header": 'Population: ', "text": "200,000"},
							{"header": 'Language: ', "text": "Galactic Basic"}
						] 
	        }
	const mockVehicle = {
	          "name": "Sand Cralwer",
	          "id": "Sand Cralwer",
	          "isFavorite": false, 
	          "type":"vehicles",
	          "properties": [
							{"header": 'Model: ', "text": "Digger Crawler"},
							{"header": 'Class: ', "text": "wheeled"},
							{"header": 'Passengers: ', "text": "1"}
						] 
	        }
	const mockPlanet = {
	          "name": "Dagobah",
	          "id": "Dagobah",
	          "isFavorite": false, 
	          "type":"planets",
	          "properties": [
							{"header": 'Terrain: ', "text": "swamp, jungle"},
							{"header": 'Population: ', "text": "unknown"},
							{"header": 'Climate: ', "text": "murky"},
							{"header": 'Residents: ', "text": ""}
						] 
	        }
	mockFavs = [{
          name: "Dina Caraballo",
          id: "Dina Caraballo",
          isFavorite: true, 
          type:"people",
          properties: [
						{header: 'Homeworld: ', text: "Earth"},
						{header: 'Species: ', text: "human"},
						{header: 'Population: ', text: "7,000,000,000"},
						{header: 'Language: ', text: "Galactic Basic"}
					] 
        },
        {
          name: "Alex Ela",
          id: "Alex Ela",
          isFavorite: true, 
          type:"people",
          properties: [
						{header: 'Homeworld: ', text: "Earth"},
						{header: 'Species: ', text: "human"},
						{header: 'Population: ', text: "7,000,000,000"},
						{header: 'Language: ', text: "Galactic Basic"}
					] 
        }, 
        {
          "name": "Luke Skywalker",
          "id": "Luke Skywalker",
          "isFavorite": true, 
          "type":"people",
          "properties": [
						{"header": 'Homeworld: ', "text": "Tatooine"},
						{"header": 'Species: ', "text": "human"},
						{"header": 'Population: ', "text": "200,000"},
						{"header": 'Language: ', "text": "Galactic Basic"}
						] 
	        }]

	beforeEach(() => {
		mockEntry = mockPerson
		wrapper = shallow(<Card entry={mockEntry} 
			key={mockEntry.id}
			toggleFavorites={mockToggleFavs} 
			favorites={mockFavs} 
			isFavorite={mockEntry.isFavorite}/>)
	})

	it('matches the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	})

	it('renders people cards if people props are present', () => {
			expect(wrapper.find('h3').length).toEqual(1)
	})

	it('renders vehicle cards if vehicle props are present', () => {
		mockEntry = mockVehicle
		expect(wrapper.find('h3').length).toEqual(1)
	})

	it('renders planet cards if planet props are present', () => {
		mockEntry = mockPlanet
		expect(wrapper.find('h3').length).toEqual(1)
	})

	it('invokes toggleFavorites on click', () => {
		wrapper.find('.fav-btn').simulate('click')
		expect(mockToggleFavs).toHaveBeenCalled()
	})

	it('changes fav button class to active if isFavorite is true', () => {
		wrapper = shallow(<Card entry={mockEntry} 
			key={mockEntry.id}
			toggleFavorites={mockToggleFavs} 
			favorites={[]} 
			isFavorite={true}/>)
		wrapper.find('button').simulate('click')
		expect(wrapper.find('button').hasClass('fav-btn-active')).toBe(true)
	})

	it('changes fav button class to inactive if card is not in the favorites array in state', () => {
	  wrapper = shallow(<Card entry={mockEntry} 
			key={mockEntry.id}
			toggleFavorites={mockToggleFavs} 
			favorites={[]} 
			isFavorite={false}/>)

		wrapper.find('button').simulate('click')
		expect(wrapper.find('button').hasClass('fav-btn-inactive')).toBe(true)
	});

	it('changes fav button class to active if card is in the favorites array in state', () => {
	  mockEntry = {
	          "name": "Luke Skywalker",
	          "id": "Luke Skywalker",
	          "isFavorite": true, 
	          "type":"people",
	          "properties": [
							{"header": 'Homeworld: ', "text": "Tatooine"},
							{"header": 'Species: ', "text": "human"},
							{"header": 'Population: ', "text": "200,000"},
							{"header": 'Language: ', "text": "Galactic Basic"}
						] 
	        }

	  wrapper = shallow(<Card entry={mockEntry} 
			key={mockEntry.id}
			toggleFavorites={mockToggleFavs} 
			favorites={mockFavs} 
			isFavorite={true}/>)

		wrapper.find('button').simulate('click')
		expect(wrapper.find('button').hasClass('fav-btn-active')).toBe(true)
	});
})	
