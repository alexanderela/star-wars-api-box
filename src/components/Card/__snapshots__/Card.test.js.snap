// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    entry={
      Object {
        "id": "Luke Skywalker",
        "isFavorite": false,
        "name": "Luke Skywalker",
        "properties": Array [
          Object {
            "header": "Homeworld: ",
            "text": "Tatooine",
          },
          Object {
            "header": "Species: ",
            "text": "human",
          },
          Object {
            "header": "Population: ",
            "text": "200,000",
          },
          Object {
            "header": "Language: ",
            "text": "Galactic Basic",
          },
        ],
        "type": "people",
      }
    }
    favorites={
      Array [
        Object {
          "id": "Dina Caraballo",
          "isFavorite": true,
          "name": "Dina Caraballo",
          "properties": Array [
            Object {
              "header": "Homeworld: ",
              "text": "Earth",
            },
            Object {
              "header": "Species: ",
              "text": "human",
            },
            Object {
              "header": "Population: ",
              "text": "7,000,000,000",
            },
            Object {
              "header": "Language: ",
              "text": "Galactic Basic",
            },
          ],
          "type": "people",
        },
        Object {
          "id": "Alex Ela",
          "isFavorite": true,
          "name": "Alex Ela",
          "properties": Array [
            Object {
              "header": "Homeworld: ",
              "text": "Earth",
            },
            Object {
              "header": "Species: ",
              "text": "human",
            },
            Object {
              "header": "Population: ",
              "text": "7,000,000,000",
            },
            Object {
              "header": "Language: ",
              "text": "Galactic Basic",
            },
          ],
          "type": "people",
        },
        Object {
          "id": "Luke Skywalker",
          "isFavorite": true,
          "name": "Luke Skywalker",
          "properties": Array [
            Object {
              "header": "Homeworld: ",
              "text": "Tatooine",
            },
            Object {
              "header": "Species: ",
              "text": "human",
            },
            Object {
              "header": "Population: ",
              "text": "200,000",
            },
            Object {
              "header": "Language: ",
              "text": "Galactic Basic",
            },
          ],
          "type": "people",
        },
      ]
    }
    isFavorite={false}
    toggleFavorites={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fav-btn-card-container"
        >
          <h3>
            Luke Skywalker
          </h3>
          <button
            className="fav-btn 
						fav-btn-inactive"
            onClick={[Function]}
            type="submit"
          >
            <i
              className="fas fa-jedi"
            />
          </button>
        </div>,
        Array [
          <p
            className="card-text"
          >
            <span
              className="card-header"
            >
              Homeworld: 
            </span>
            Tatooine
          </p>,
          <p
            className="card-text"
          >
            <span
              className="card-header"
            >
              Species: 
            </span>
            human
          </p>,
          <p
            className="card-text"
          >
            <span
              className="card-header"
            >
              Population: 
            </span>
            200,000
          </p>,
          <p
            className="card-text"
          >
            <span
              className="card-header"
            >
              Language: 
            </span>
            Galactic Basic
          </p>,
        ],
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Luke Skywalker
            </h3>,
            <button
              className="fav-btn 
						fav-btn-inactive"
              onClick={[Function]}
              type="submit"
            >
              <i
                className="fas fa-jedi"
              />
            </button>,
          ],
          "className": "fav-btn-card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Luke Skywalker",
            },
            "ref": null,
            "rendered": "Luke Skywalker",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-jedi"
              />,
              "className": "fav-btn 
						fav-btn-inactive",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-jedi",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "Homeworld: ",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="card-header"
            >
              Homeworld: 
            </span>,
            "Tatooine",
          ],
          "className": "card-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Homeworld: ",
              "className": "card-header",
            },
            "ref": null,
            "rendered": "Homeworld: ",
            "type": "span",
          },
          "Tatooine",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "Species: ",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="card-header"
            >
              Species: 
            </span>,
            "human",
          ],
          "className": "card-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Species: ",
              "className": "card-header",
            },
            "ref": null,
            "rendered": "Species: ",
            "type": "span",
          },
          "human",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "Population: ",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="card-header"
            >
              Population: 
            </span>,
            "200,000",
          ],
          "className": "card-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Population: ",
              "className": "card-header",
            },
            "ref": null,
            "rendered": "Population: ",
            "type": "span",
          },
          "200,000",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "Language: ",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="card-header"
            >
              Language: 
            </span>,
            "Galactic Basic",
          ],
          "className": "card-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Language: ",
              "className": "card-header",
            },
            "ref": null,
            "rendered": "Language: ",
            "type": "span",
          },
          "Galactic Basic",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fav-btn-card-container"
          >
            <h3>
              Luke Skywalker
            </h3>
            <button
              className="fav-btn 
						fav-btn-inactive"
              onClick={[Function]}
              type="submit"
            >
              <i
                className="fas fa-jedi"
              />
            </button>
          </div>,
          Array [
            <p
              className="card-text"
            >
              <span
                className="card-header"
              >
                Homeworld: 
              </span>
              Tatooine
            </p>,
            <p
              className="card-text"
            >
              <span
                className="card-header"
              >
                Species: 
              </span>
              human
            </p>,
            <p
              className="card-text"
            >
              <span
                className="card-header"
              >
                Population: 
              </span>
              200,000
            </p>,
            <p
              className="card-text"
            >
              <span
                className="card-header"
              >
                Language: 
              </span>
              Galactic Basic
            </p>,
          ],
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Luke Skywalker
              </h3>,
              <button
                className="fav-btn 
						fav-btn-inactive"
                onClick={[Function]}
                type="submit"
              >
                <i
                  className="fas fa-jedi"
                />
              </button>,
            ],
            "className": "fav-btn-card-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Luke Skywalker",
              },
              "ref": null,
              "rendered": "Luke Skywalker",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-jedi"
                />,
                "className": "fav-btn 
						fav-btn-inactive",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-jedi",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Homeworld: ",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-header"
              >
                Homeworld: 
              </span>,
              "Tatooine",
            ],
            "className": "card-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Homeworld: ",
                "className": "card-header",
              },
              "ref": null,
              "rendered": "Homeworld: ",
              "type": "span",
            },
            "Tatooine",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "Species: ",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-header"
              >
                Species: 
              </span>,
              "human",
            ],
            "className": "card-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Species: ",
                "className": "card-header",
              },
              "ref": null,
              "rendered": "Species: ",
              "type": "span",
            },
            "human",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "Population: ",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-header"
              >
                Population: 
              </span>,
              "200,000",
            ],
            "className": "card-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Population: ",
                "className": "card-header",
              },
              "ref": null,
              "rendered": "Population: ",
              "type": "span",
            },
            "200,000",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "Language: ",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-header"
              >
                Language: 
              </span>,
              "Galactic Basic",
            ],
            "className": "card-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Language: ",
                "className": "card-header",
              },
              "ref": null,
              "rendered": "Language: ",
              "type": "span",
            },
            "Galactic Basic",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
